protected/humhub/modules/content/search/driver/MysqlDriver.php

```diff
@@ -172,10 +172,16 @@ protected function addQueryFilterVisibility(ActiveQuery $query): ActiveQuery
     {
         $query->andWhere(['content.state' => Content::STATE_PUBLISHED]);
 
-        $query->joinWith('contentContainer');
+        $query->joinWith(['contentContainer', 'createdBy']);
         $query->leftJoin('space', 'contentcontainer . pk = space . id and contentcontainer .class=:spaceClass', [':spaceClass' => Space::class]);
         $query->leftJoin('user cuser', 'contentcontainer . pk = cuser . id and contentcontainer .class=:userClass', [':userClass' => User::class]);
 
+        // Filter out content created by not enabled users
+        $query->andWhere(['OR',
```

Comment:
I don't think we can realise this for other search engines, as a simple join is not possible here.
Would an ActiveJob (RemoveUserContent, AddUserContent) be a solution here?

Reply:
@luke- Maybe we should not index content if author status is not enabled, we could update the method [ContentSearchService::isIndexable()](https://github.com/humhub/humhub/blob/fix/315-disabled-user-content-readable/protected/humhub/modules/content/services/ContentSearchService.php#L81-L84) to:

```php
    public function isIndexable(): bool
    {
        return $this->content->stream_channel === Content::STREAM_CHANNEL_DEFAULT &&
            $this->content->createdBy->status === User::STATUS_ENABLED;
    }
```

Reply:
@luke- 

> But what should we do with already indexed user content?

Nothing, I think only next rebuilding will solve it. It can be solved by Mysql driver, but not for ZendLucence Driver.

Reply:
Commit https://github.com/humhub/humhub/pull/7153/commits/60e9723075583069768b5c493bed3cec21a1901d for new changes.

Reply:
@yurabakhtin Looks good. Another point would be an ActiveJob "ReindexUserContent" which deletes indexed content using `created_by` and reindex it?